Makefile

```diff
@@ -43,12 +45,15 @@ all: apps.img
 	cp apps.img $(FROSTED)/
 
 
-apps.img: $(APPS-y) $(DIR-y)
-	@mkdir -p gdb
-	@mv binutils/out/*.gdb $(PWD)/gdb || true
-	@mv hw-utils/out/*.gdb $(PWD)/gdb || true
-	@mv netutils/out/*.gdb $(PWD)/gdb || true
-	$(FROSTED)/tools/xipfstool $@ $(APPS-y) $(addsuffix /out/*,$(DIR-y))
+apps.img: $(APPS-y) $(DIR-y) $(OUT-y) $(GDB-y)
+	@mkdir -p $(PWD)/lib/out
+	@find binutils/out/* -type f -executable | grep -v "gdb" | xargs cp -t $(OUT) || true 
+	@find netutils/out/* -type f -executable | grep -v "gdb" | xargs cp -t $(OUT) || true 
+	@find netutils/out/* -type f -executable | grep -v "gdb" | xargs cp -t $(OUT) || true
+	@find binutils/out/* -type f -executable | grep "gdb" | xargs cp -t $(GDB) || true 
+	@find hw-utils/out/* -type f -executable | grep "gdb" | xargs cp -t $(GDB) || true 
+	@find netutils/out/* -type f -executable | grep "gdb" | xargs cp -t $(GDB) || true
+	$(FROSTED)/tools/xipfstool $@ $(APPS-y) $(DIR-y)
```

Comment:
stuff in extra/ should also become part of the xipfs distribution: try putting a readme.txt in extra and ensure it gets in the xipfs?